lib/main.js

```diff
@@ -3,96 +3,174 @@
 const url = require('url');
 const JSONStream = require('JSONStream');
 const request = require('request');
-const Writer = require('asset-pipe-js-writer');
+const JsWriter = require('asset-pipe-js-writer');
+const CssWriter = require('asset-pipe-css-writer');
+const assert = require('assert');
+const { extname } = require('path');
+const { Stream } = require('stream');
+
+const extensions = Object.freeze({
+    JS: '.js',
+    CSS: '.css',
+});
+
+const endpoints = Object.freeze({
+    UPLOAD: 'feed',
+    BUNDLE: 'bundle',
+});
+
+function post(options) {
+    const isStream = options.body instanceof Stream;
+    const opts = {
+        url: options.url,
+        headers: {
```

Comment:
we should add a reasonable timeout here?

Reply:
oh good point. How long do you think is reasonable?

Reply:
We did wrap this with timeout at 500 and retries when using the client. Maybe the client should support retries(with `p-retry`) and e.g. 5000 seconds as timeout out of the box?

Reply:
how would this work in a streaming context? If I pipe the writer stream into request the stream will be emptied before a retry is possible right?